join/join_helpers/join_helpers.hpp

```diff
@@ -0,0 +1,130 @@
+#pragma once
+#include "common/common.hpp"
+
+namespace join_helpers {
+
+template <class Key, class Val1, class Val2>
+using ColJoinedTableTy =
+    std::pair<std::vector<Key>,
+              std::pair<std::vector<Val1>, std::vector<Val2>>>;
+
+template <class Key, class V1, class V2>
+using RowJoinedTableTy = std::vector<std::pair<Key, std::pair<V1, V2>>>;
+
+template <class K, class V1, class V2>
+bool is_malformed(const ColJoinedTableTy<K, V1, V2> &t) {
+  return t.first.size() != t.second.first.size() ||
+         t.first.size() != t.second.second.size();
+}
+
+template <class K, class V1, class V2>
+size_t get_size(const ColJoinedTableTy<K, V1, V2> &t) {
+  if (is_malformed(t))
+    throw std::invalid_argument("ColJoinedTableTy is malformed.");
```

Comment:
How should your code recover from such an exception?

Reply:
Is this my code btw? :)

Reply:
yes, this is your code :D
I just took it out
and what's the problem again? I didn't quite get it. the question is how do I handle exception in join?